lib/wp-taxonomy-filter/class-cf-taxonomy-filter.php

```diff
@@ -0,0 +1,426 @@
+<?php
+/**
+ * Class implementing main functionality for taxonomy filter.
+ *
+ * @package cf-taxonomy-filter
+ */
+
+/**
+ * Taxonomy filter class.
+ */
+class CF_Taxonomy_Filter {
+	/**
+	 * Constructor.
+	 *
+	 * @param array $args Additional arguments.
+	 */
+	public function __construct( $args ) {
+		// These keys are always required. Set them here so we don't have to think about existence later.
+		$default_keys  = array(
+			'form_options'   => array(),
+			'submit_options' => array(),
+		);
+		$this->options = array_merge( $default_keys, $args );
+	}
+
+	/**
+	 * Add actions hook.
+	 *
+	 * @return void
+	 */
+	public static function add_actions() {
+		add_action( 'pre_get_posts', array( 'CF_Taxonomy_Filter', 'pre_get_posts' ) );
+	}
+
+	/**
+	 * Builds a form based on arguments passed into the constructor.
+	 *
+	 * @return void
+	 */
+	public function build_form() {
+		self::start_form( $this->options['form_options'] );
+
+		if ( ! empty( $this->options['taxonomies'] ) ) {
+			foreach ( $this->options['taxonomies'] as $taxonomy => $args ) {
+				if ( is_array( $args ) ) {
+					self::tax_filter( $taxonomy, $args );
+				} // Just passed in taxonomy name with no options
+				else {
+					self::tax_filter( $args );
+				}
+			}
+		}
+
+		if ( ! empty( $this->options['authors'] ) ) {
+			$author_options = ! empty( $this->options['author_options'] ) ? $this->options['author_options'] : array();
+			self::author_select( $author_options );
+		}
+
+		if ( ! empty( $this->options['date'] ) ) {
+			$start_options = ! empty( $this->options['date_options']['start'] ) ? $this->options['date_options']['start'] : array();
+			$end_options   = ! empty( $this->options['date_options']['end'] ) ? $this->options['date_options']['end'] : array();
+			self::date_filter( $start_options, $end_options );
+		}
+
+		self::submit_button( $this->options['submit_options'] );
+
+		self::end_form();
+	}
+
+	/**
+	 * Echo a date range filter form element
```

Comment:
Missing . at the end of the line.

Reply:
Updating. However latest phpcs does not report this warning for method descriptions comments

Reply:
Oh, I see. Still nice to be consistent with inline comments, though :) But not a biggie...